quantmsrescore/model_downloader.py

```
@@ -0,0 +1,442 @@
+    try:
+        model_dir = model_dir if model_dir else Path.home() / ".ms2pip"
+        model_dir = Path(model_dir).expanduser()
+        model_dir.mkdir(parents=True, exist_ok=True)
+        logger.info(f"Available MS2PIP models: {list(MODELS.keys())}")
+        models = list(MODELS.keys())
+        for model in models:
+            try:
+                if "xgb_model_files" in MODELS[model].keys():
+                    continue
+            except KeyError:
+                raise exceptions.UnknownModelError(model)
```

Ruff (0.14.7)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)